### ISSUE TITLE
IllegalArgumentException getting page

### ISSUE THREAD

**Issue by ares242** [tyl/mongodbcontainer-addon#7]
Some times in production mode this exception is thrown, I am not sure but it happens when I scroll quickly from bottom to top 
```java
2015-02-04 13:38:19,685 INFO  [MongoContainer] (default task-5) range: [55,18]
2015-02-04 13:38:19,686 SEVERE [com.vaadin.server.DefaultErrorHandler] (default task-5) : java.lang.IllegalArgumentException: fromIndex(55) > toIndex(18)
	at java.util.SubList.<init>(AbstractList.java:624) [rt.jar:1.8.0_31]
	at java.util.RandomAccessSubList.<init>(AbstractList.java:775) [rt.jar:1.8.0_31]
	at java.util.AbstractList.subList(AbstractList.java:484) [rt.jar:1.8.0_31]
	at java.util.Collections$UnmodifiableRandomAccessList.subList(Collections.java:1400) [rt.jar:1.8.0_31]
	at org.tylproject.vaadin.addon.utils.Page.subList(Page.java:134) [mongodbcontainer-addon-0.9.5.6-beta.jar:0.9.5.6-beta]
	at org.tylproject.vaadin.addon.MongoContainer.getItemIds(MongoContainer.java:501) [mongodbcontainer-addon-0.9.5.6-beta.jar:0.9.5.6-beta]
	at com.vaadin.ui.Table.getItemIds(Table.java:2241) [vaadin-server-7.4.0.beta3.jar:7.4.0.beta3]
	at com.vaadin.ui.Table.getVisibleCellsNoCache(Table.java:2191) [vaadin-server-7.4.0.beta3.jar:7.4.0.beta3]
	at com.vaadin.ui.Table.refreshRenderedCells(Table.java:1717) [vaadin-server-7.4.0.beta3.jar:7.4.0.beta3]
	at com.vaadin.ui.Table.enableContentRefreshing(Table.java:3242) [vaadin-server-7.4.0.beta3.jar:7.4.0.beta3]
	at com.vaadin.ui.Table.changeVariables(Table.java:3088) [vaadin-server-7.4.0.beta3.jar:7.4.0.beta3]
	at com.vaadin.server.communication.ServerRpcHandler.changeVariables(ServerRpcHandler.java:486) [vaadin-server-7.4.0.beta3.jar:7.4.0.beta3]
	at com.vaadin.server.communication.ServerRpcHandler.handleInvocations(ServerRpcHandler.java:305) [vaadin-server-7.4.0.beta3.jar:7.4.0.beta3]
	at com.vaadin.server.communication.ServerRpcHandler.handleRpc(ServerRpcHandler.java:184) [vaadin-server-7.4.0.beta3.jar:7.4.0.beta3]
	at com.vaadin.server.communication.PushHandler$3.run(PushHandler.java:174) [vaadin-server-7.4.0.beta3.jar:7.4.0.beta3]
	at com.vaadin.server.communication.PushHandler.callWithUi(PushHandler.java:253) [vaadin-server-7.4.0.beta3.jar:7.4.0.beta3]
	at com.vaadin.server.communication.PushHandler.access$200(PushHandler.java:56) [vaadin-server-7.4.0.beta3.jar:7.4.0.beta3]
	at com.vaadin.server.communication.PushHandler$1.onRequest(PushHandler.java:76) [vaadin-server-7.4.0.beta3.jar:7.4.0.beta3]
	at org.atmosphere.cpr.AsynchronousProcessor.action(AsynchronousProcessor.java:205) [atmosphere-runtime-2.2.4.vaadin2.jar:2.2.4.vaadin2]
	at org.atmosphere.cpr.AsynchronousProcessor.suspended(AsynchronousProcessor.java:104) [atmosphere-runtime-2.2.4.vaadin2.jar:2.2.4.vaadin2]
	at org.atmosphere.container.Servlet30CometSupport.service(Servlet30CometSupport.java:66) [atmosphere-runtime-2.2.4.vaadin2.jar:2.2.4.vaadin2]
	at org.atmosphere.cpr.AtmosphereFramework.doCometSupport(AtmosphereFramework.java:2075) [atmosphere-runtime-2.2.4.vaadin2.jar:2.2.4.vaadin2]
	at org.atmosphere.websocket.DefaultWebSocketProcessor.dispatch(DefaultWebSocketProcessor.java:570) [atmosphere-runtime-2.2.4.vaadin2.jar:2.2.4.vaadin2]
	at org.atmosphere.websocket.DefaultWebSocketProcessor$3.run(DefaultWebSocketProcessor.java:332) [atmosphere-runtime-2.2.4.vaadin2.jar:2.2.4.vaadin2]
	at org.atmosphere.util.VoidExecutorService.execute(VoidExecutorService.java:101) [atmosphere-runtime-2.2.4.vaadin2.jar:2.2.4.vaadin2]
	at org.atmosphere.websocket.DefaultWebSocketProcessor.dispatch(DefaultWebSocketProcessor.java:327) [atmosphere-runtime-2.2.4.vaadin2.jar:2.2.4.vaadin2]
	at org.atmosphere.websocket.DefaultWebSocketProcessor.invokeWebSocketProtocol(DefaultWebSocketProcessor.java:424) [atmosphere-runtime-2.2.4.vaadin2.jar:2.2.4.vaadin2]
	at org.atmosphere.container.JSR356Endpoint$1.onMessage(JSR356Endpoint.java:206) [atmosphere-runtime-2.2.4.vaadin2.jar:2.2.4.vaadin2]
	at org.atmosphere.container.JSR356Endpoint$1.onMessage(JSR356Endpoint.java:203) [atmosphere-runtime-2.2.4.vaadin2.jar:2.2.4.vaadin2]
	at io.undertow.websockets.jsr.FrameHandler$7.run(FrameHandler.java:249) [undertow-websockets-jsr-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.websockets.jsr.ServerWebSocketContainer.invokeEndpointMethod(ServerWebSocketContainer.java:362) [undertow-websockets-jsr-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.websockets.jsr.ServerWebSocketContainer$3.run(ServerWebSocketContainer.java:347) [undertow-websockets-jsr-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.websockets.jsr.OrderedExecutor$ExecutorTask.run(OrderedExecutor.java:67) [undertow-websockets-jsr-1.1.0.Final.jar:1.1.0.Final]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [rt.jar:1.8.0_31]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [rt.jar:1.8.0_31]
	at java.lang.Thread.run(Thread.java:745) [rt.jar:1.8.0_31]
```

**Comment by evacchi** [comment]
Hi,

I have just uploaded v0.9.6, and I can't seem to be able to reproduce this. Can you try with this version and report back if the bug is still there?

thanks

**Comment by ares242** [comment]
I have tried with the new version and it keeps happening, I forgot to mention that it happened when I added the @Push vaadin annotation to the UI. Now I have been testing MongoDB Container with new vaadin feature grid and in the grid initialization it throws a similar error:

```java
Feb 10, 2015 12:39:10 PM org.tylproject.vaadin.addon.MongoContainer getItemIds
INFO: range: [50,40]
Feb 10, 2015 12:39:10 PM com.vaadin.server.DefaultErrorHandler doDefault
SEVERE: 
java.lang.IllegalArgumentException: fromIndex(50) > toIndex(40)
	at java.util.SubList.<init>(AbstractList.java:624)
	at java.util.RandomAccessSubList.<init>(AbstractList.java:775)
	at java.util.RandomAccessSubList.subList(AbstractList.java:779)
	at java.util.Collections$UnmodifiableRandomAccessList.subList(Collections.java:1400)
	at org.tylproject.vaadin.addon.utils.Page.subList(Page.java:135)
	at org.tylproject.vaadin.addon.MongoContainer.getItemIds(MongoContainer.java:528)
	at com.vaadin.data.RpcDataProviderExtension.pushRowData(RpcDataProviderExtension.java:772)
	at com.vaadin.data.RpcDataProviderExtension.access$1100(RpcDataProviderExtension.java:74)
	at com.vaadin.data.RpcDataProviderExtension$2.requestRows(RpcDataProviderExtension.java:709)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at com.vaadin.server.ServerRpcManager.applyInvocation(ServerRpcManager.java:168)
	at com.vaadin.server.ServerRpcManager.applyInvocation(ServerRpcManager.java:118)
	at com.vaadin.server.communication.ServerRpcHandler.handleInvocations(ServerRpcHandler.java:291)
	at com.vaadin.server.communication.ServerRpcHandler.handleRpc(ServerRpcHandler.java:184)
	at com.vaadin.server.communication.UidlRequestHandler.synchronizedHandleRequest(UidlRequestHandler.java:92)
	at com.vaadin.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:41)
	at com.vaadin.server.VaadinService.handleRequest(VaadinService.java:1408)
	at com.vaadin.server.VaadinServlet.service(VaadinServlet.java:350)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:800)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1669)
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:224)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:61)
	at org.apache.shiro.web.servlet.AdviceFilter.executeChain(AdviceFilter.java:108)
	at org.apache.shiro.web.servlet.AdviceFilter.doFilterInternal(AdviceFilter.java:137)
	at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125)
	at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:66)
	at org.apache.shiro.web.servlet.AbstractShiroFilter.executeChain(AbstractShiroFilter.java:449)
	at org.apache.shiro.web.servlet.AbstractShiroFilter$1.call(AbstractShiroFilter.java:365)
	at org.apache.shiro.subject.support.SubjectCallable.doCall(SubjectCallable.java:90)
	at org.apache.shiro.subject.support.SubjectCallable.call(SubjectCallable.java:83)
	at org.apache.shiro.subject.support.DelegatingSubject.execute(DelegatingSubject.java:383)
	at org.apache.shiro.web.servlet.AbstractShiroFilter.doFilterInternal(AbstractShiroFilter.java:362)
	at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125)
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:344)
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:261)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:577)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:215)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:110)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
	at org.eclipse.jetty.server.Server.handle(Server.java:497)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:245)
	at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by evacchi** [comment]
Bug should be now fixed in v0.9.6.1-beta

**Comment by ares242** [comment]
Great! thank you!
